subprojects/plugin-development/src/integTest/groovy/org/gradle/plugin/devel/tasks/ValidateTaskPropertiesIntegrationTest.groovy

```diff
@@ -387,6 +390,59 @@ class ValidateTaskPropertiesIntegrationTest extends AbstractIntegrationSpec {
         file("build/reports/task-properties/report.txt").text == ""
     }
 
+
+    @Unroll
+    def "report setters for mutable type #type"() {
+        @Language("JAVA") myTask = """
+            import org.gradle.api.DefaultTask;
+            import org.gradle.api.tasks.InputFiles;
+
+            public class MyTask extends DefaultTask {
+                private ${type} sources;
+                
+                @InputFiles
+                public ${type} getSources() { return sources; } 
+                public void setSources(${type} sources) { this.sources = sources; } 
+            }
+        """
+        file("src/main/java/MyTask.java") << myTask
+
+        when:
+        fails "validateTaskProperties"
+
+        then:
+        file("build/reports/task-properties/report.txt").text == """
+            Warning: Type 'MyTask': Support for property setters for type '${type.replaceAll("<.+>", "")}' will be removed in a future release. Please consider making 'sources' final.
+        """.stripIndent().trim()
+
+        where:
+        type << [ConfigurableFileCollection.name, "${Property.name}<${String.name}>"]
+    }
+
+    @Unroll
+    def "don't report when non-final field doesn't have setters for type #type"() {
+        @Language("JAVA") myTask = """
+            import org.gradle.api.DefaultTask;
+            import org.gradle.api.tasks.InputFiles;
+
+            public class MyTask extends DefaultTask {
+                private final ${type} sources;
+                
+                MyTask() { sources = null; }
+                
+                @InputFiles
+                public ${type} getSources() { return sources; } 
+            }
+        """
+        file("src/main/java/MyTask.java") << myTask
+
+        expect:
+        succeeds "validateTaskProperties"
+
+        where:
+        type << [ConfigurableFileCollection.name, "${Property.name}<${String.name}>"]
```

Comment:
Maybe just `"Property<String>"` is fine. I mean, this is really complicated, and if we ever decide to rename `Property` (or `String` :p) updating this test will be the least of our worries.

Reply:
Done